### ISSUE TITLE
Error when downloading the default cheatsheet

### ISSUE THREAD

**Issue by Aracki** [denisidoro/navi#326]
When running `navi` after install and selecting the "Download default cheatssheets" will cause the following error:
```
Cloning https://github.com/denisidoro/cheats into /Users/<user>/Library/Application Support/navi/cheats/tmp...

Hey listen! Navi encountered a problem.
Do you think this is a bug? File an issue at https://github.com/denisidoro/navi.

Caused by:
    0: Failed to import cheatsheets from `denisidoro/cheats`
    1: Failed to clone `https://github.com/denisidoro/cheats`
    2: unsupported URL protocol; class=Net (12)
```

**Comment by denisidoro** [comment]
Maybe related? https://github.com/denisidoro/navi/issues/292#issuecomment-601711023

**Comment by ryuta69** [comment]
This happens to me as well with latest version.

**Comment by ryuta69** [comment]
ah ok it happens because this exists

```
[url "ssh://git@github.com/"]
       insteadOf = https://github.com/
```

It might be annoying, so considering to PR.

**Comment by ryuta69** [comment]
maybe relate to this one.

https://github.com/denisidoro/navi/pull/277

I'll look into it this evening(it's morning now in japan).

**Comment by denisidoro** [comment]
Probably fixed by https://github.com/denisidoro/navi/pull/336

If this error persists after I release the 2.5+ version, please feel free to reopen the issue.

**Comment by Aracki** [comment]
In 2.5 version, after selecting "Download default cheatsheets" I am getting 0/0 screen.
<img width="488" alt="image" src="https://user-images.githubusercontent.com/6129986/79064315-25dfdf00-7ca8-11ea-87f5-94bedfe52a6e.png">

**Comment by denisidoro** [comment]
@Aracki were there any previous logs? Something like "attempting to clone... into..."?

**Comment by denisidoro** [comment]
I tested the 2.5 version on a vanilla Ubuntu machine and it worked: https://www.katacoda.com/denisidoro/scenarios/navi

If gotta be something related to git or the filesystem :thinking:

**Comment by Aracki** [comment]
@denisidoro Nope 🙁 Using `MacOS 10.15.3`

**Comment by denisidoro** [comment]
When you git clone a repo (outside navi), does it prompt you for a password?

**Comment by Aracki** [comment]
No

**Comment by EItanya** [comment]
I am running into this issue as well on `MacOS 10.15.2`. From what I can tell it has to do with the default location `navi` uses to store cheat sheets. According to the [docs](https://github.com/denisidoro/navi#adding-your-own-cheatsheets) it uses the `rust` `data_dir` which on mac points to `Some($HOME/Library/Application Support)`. 
When running `navi repo add` nothing showed up for me as well. However, when I `ctrl+c` I get this output in the terminal
```
Cloning https://github.com/denisidoro/cheats into /Users/<omitted>/Library/Application Support/navi/cheats/tmp...

fatal: Too many arguments.
```
After some investigation it turns out the the `repo add` command [here](https://github.com/denisidoro/navi/blob/4451624ce0179823c9bd616362f24f2dc75501b8/src/flows/repo.rs#L46) gets the directory from the [`cheat_pathbuf`](https://github.com/denisidoro/navi/blob/4451624ce0179823c9bd616362f24f2dc75501b8/src/filesystem.rs#L30) function which outputs multi-word path elements without the escape elements. 

To fix this issue the path returned by [`pathbuf_to_string`](https://github.com/denisidoro/navi/blob/4451624ce0179823c9bd616362f24f2dc75501b8/src/filesystem.rs#L22) should be escaped.

Happy to help if you agree with this approach @denisidoro

**Comment by denisidoro** [comment]
@EItanya review please?
https://github.com/denisidoro/navi/pull/340

**Comment by EItanya** [comment]
Verified that it worked locally. 👍 
Thanks so much!

**Comment by denisidoro** [comment]
Thanks! I still need to fix some bugs to release 2.5.1 with the fixes but hopefully I'll do it by tomorrow.
